### PR TITLE
(minor) Add backticks around type names

### PR DESCRIPTION
**Pull request by JOE1994**
Added backticks around type names for consistency of document.

### CODE REVIEW

**Comment by JOE1994** [comment]
CI check is currently failing due to a separate issue from this commit

**Comment by lzutao** [comment]
Will be fixed by #188

File: src/subtyping.md [hunk 1]
```diff
@@ -183,8 +183,8 @@ to trying to explain:
 The types with \*'s are the ones we will be focusing on, as they are in
 some sense "fundamental". All the others can be understood by analogy to the others:
 
-* Vec and all other owning pointers and collections follow the same logic as Box
-* Cell and all other interior mutability types follow the same logic as UnsafeCell
+* `Vec` and all other owning pointers and collections follow the same logic as `Box`
```
Comment (Centril) [inline src/subtyping.md]:
```suggestion
* `Vec<T>` and all other owning pointers and collections follow the same logic as `Box<T>`
```

File: src/subtyping.md [hunk 2]
```diff
@@ -183,8 +183,8 @@ to trying to explain:
...
+* `Cell` and all other interior mutability types follow the same logic as `UnsafeCell`
```
Comment (Centril) [inline src/subtyping.md]:
```suggestion
* `Cell<T>` and all other interior mutability types follow the same logic as `UnsafeCell<T>`
```

File: src/subtyping.md [hunk 3]
```diff
@@ -183,8 +183,8 @@ to trying to explain:
...
 * `*const` follows the logic of `&T`
```
Comment (Centril) [inline src/subtyping.md]:
```suggestion
* `*const T` follows the logic of `&T`
```

File: src/subtyping.md [hunk 4]
```diff
@@ -183,8 +183,8 @@ to trying to explain:
...
 * `*mut` follows the logic of `&mut T` (or `UnsafeCell<T>`)
```
Comment (Centril) [inline src/subtyping.md]:
```suggestion
* `*mut T` follows the logic of `&mut T` (or `UnsafeCell<T>`)
```

**Pull request merged**